app/src/main/java/com/huji/foodtricks/buddies/RSVPListAdapter.java

```diff
@@ -0,0 +1,128 @@
+package com.huji.foodtricks.buddies;
+
+import android.app.Activity;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.ImageView;
+import android.widget.ListView;
+import android.widget.TextView;
+
+import com.bumptech.glide.request.RequestOptions;
+import com.huji.foodtricks.buddies.Models.UserModel;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Objects;
+
+public class RSVPListAdapter extends ArrayAdapter {
+
+
+    //to reference the Activity
+    private final Activity context;
+
+    //to store the animal images
+    private final UserModel[] UserModels;
+
+    //to store the list of countries
+    private final String[] nameArray;
+
+    //to store the list of countries
+    private final String[] RSVPArray;
+
+    public RSVPListAdapter(Activity context, String[] nameArrayParam, String[] RSVPArray, UserModel[] userModels) {
+
+        super(context, R.layout.layout, nameArrayParam);
+
+
+        this.context = context;
+        this.UserModels = userModels;
+        this.nameArray = nameArrayParam;
+        this.RSVPArray = RSVPArray;
+
+    }
+
+
+    private static void getUsersFromDB(UsersRSVPListAdapterCompletion completion) {
+        DatabaseStreamer dbs = new DatabaseStreamer();
+        dbs.fetchAllUsersInDBMap(completion);
+    }
+
+    public static RSVPListAdapter setupUserList(Activity context,EventAttendanceProvider eventAttendanceProvider) {
+
+        getUsersFromDB(new UsersRSVPListAdapterCompletion() {
+            @Override
+            public RSVPListAdapter onFetchSuccess(HashMap<String, UserModel> usersMap) {
+                RSVPListAdapter rsvpListAdapter = buildRSVPListAdapter(context,eventAttendanceProvider, usersMap);
+                ListView listView = context.findViewById(R.id.rsvp_listview);
+                listView.setAdapter(rsvpListAdapter);
+                return rsvpListAdapter;
+            }
+
+            @Override
+            public void onNoUsersFound() {
+                // TODO: show that there are no users?
+
+            }
+        });
+        return null;
+    }
+
+    public static RSVPListAdapter buildRSVPListAdapter(Activity context,EventAttendanceProvider eventAttendanceProvider, HashMap<String, UserModel> usersMap) {
+        ArrayList<String> names = new ArrayList<>();
+        ArrayList<String> RSVPs = new ArrayList<>();
+        ArrayList<UserModel> userModels = new ArrayList<>();
+        for (String userId : usersMap.keySet()) {
+            if (eventAttendanceProvider.getInvitees().keySet().contains(userId)) {
+                UserModel userModel = usersMap.get(userId);
+                names.add(userModel.getUserName());
+                userModels.add(userModel);
+                EventAttendanceProvider.RSVP rsvp = eventAttendanceProvider.getUserRSVP(userId);
+                switch (rsvp) {
```

Comment:
Why not stick to the shorter terminology like at the bottom of the screen? (Going/Coming, Not Going/Not Coming, Maybe, Unknown/Invited/Didn't Respond)